### ISSUE TITLE
Better support for symlinks in models directory

### ISSUE THREAD

**Issue by remixer-dec** [lmstudio-ai/lmstudio-bug-tracker#97]
I store some of my models on the main drive and some of them on an external drive, to keep them all in the same place, I use symlinks. When the drive is disconnected, an infinite loop of alert windows appears, asking me to change the models directory. The models directory still exists and it is non-empty, the symlink to my main drive still exists and points to the models on the main drive. 

Expected behavior:
Symlinks that cannot be accessed are ignored, or an alert about their status appears once.

Current behavior:
A single inaccessible symlink prevents to use the application showing an infinite loop of alert windows.

<img width="321" alt="screenshot" src="https://github.com/lmstudio-ai/lmstudio-bug-tracker/assets/6587642/b1a10a3b-5f9f-45de-ab53-c569ba0917b8">

**Comment by remixer-dec** [comment]
The issue can be fixed with 2 lines
```js
const symlinkExists = (path) => fs.existsSync(path) && fs.lstatSync(path).isSymbolicLink();

/* inside async indexModels() */
// change the line with n.isDirectory to
if (n.isDirectory() || symlinkExists(p.default.join(_, n.name))) /* ... */
```

**Comment by remixer-dec** [comment]
Also a big UX downgrade for my usecase is that in v0.3 it shows the parent directory name instead of actual model name, until you actually click it. I tried as a workaround to move the symlinked dirs into the parent folder, but then they are just ignored as two nested directories are expected.
<img width="478" alt="image" src="https://github.com/user-attachments/assets/c52ede52-ccb9-4383-91df-6eabc98a7f92">  
<img width="213" alt="image" src="https://github.com/user-attachments/assets/70277aeb-9d57-4e2a-92c1-fb469c35d3c7">

<img width="478" alt="image" src="https://github.com/user-attachments/assets/ba6b3390-bd3a-4e9a-8c6f-fc40a05dfbda">

Honestly, I am disappointed with version 0.3, the model vs model side-by-side feature is gone, the chat message box margins are extremely large and the chat boxes have hard corners while the other interfaces have round corners, user role name and user message blend into one, in my opinion the chat ui became worse, it would be nice to have the old theme available. 

Also there is a bug with JSON formatting in the new version, possibly in chat migration logic.
<img width="353" alt="image" src="https://github.com/user-attachments/assets/c0af4cdc-1955-4896-a764-f0f61da88a58">
P.S. after using it for a while, maybe the new design isn't that bad, at least with the dark theme.

**Comment by huyz** [comment]
Please fix the handling of symlinks in the models and the cache directory.
When we're dealing with very large directories, users need to be able to use symlinks to manage their disk space.
So please:
1) follow symlinks
2) if symlinks point to nowhere, ignore them.

See also:
- https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/216
- https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/343#issuecomment-2614156289
- https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/294#issuecomment-2585349175
- https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/294#issuecomment-2635438579

**Comment by remixer-dec** [comment]
thanks for obfuscating the app code and NOT adding symlinks functionality, now people have to waste hours to add it on their local machine

and thanks for contributing to global warming by doing millions of unnecessary function calls just to obfuscate the code

**Comment by aaaronmiller** [comment]
WTF why hasn't this issue been solved yet ? these models are huge; not correctly handling symlinks is unacceptable! get it together already! such a hassle! expected behavior: if symlinks are present; show the models. if symlinks are broken; don't show (but show if the external drive is later reconnected, duhh)


EDIT: DUhhhhh.. you just need to symlink the actual model files; not the directory itself and it works fine. bah-humbug

**Comment by hle-pc** [comment]
@aaaronmiller yes that's what I've been doing. But it's inconvenient to do repeatedly. And an unconventional constraint regardless

**Comment by remixer-dec** [comment]
Seems like symlinks are working correctly now in the latest versions of LM Studio, thank you!
